### PR TITLE
Bump dependencies to latest version

### LINKED ISSUE #249: Bump WPCS to 3.0.0
### Is your enhancement related to a problem? Please describe.

The latest version of [WPCS 3.0.0](https://make.wordpress.org/core/2023/08/21/wordpresscs-3-0-0-is-now-available/) is released yesterday. We should update it so we can remove some development version dependancies and more sniffs.

### Designs

_No response_

### Describe alternatives you've considered

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by mukeshpanchal27**
This PR updates composer dependancies.
- Update WPCS to latest 3.0 development version commit hash.
- Update [Automattic/VIP-Coding-Standards](https://github.com/Automattic/VIP-Coding-Standards) to latest version
- Correct fail unit test.

Closes #249

### Checklist:

- [x] I agree to follow this project's [**Code of Conduct**](https://github.com/10up/.github/blob/trunk/CODE_OF_CONDUCT.md).

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -7,8 +7,8 @@
     "php": ">=5.6",
     "composer/installers": "^v1.12.0 || ^2.2",
     "dealerdirect/phpcodesniffer-composer-installer": "^0.4.1 || ^0.5 || ^0.6.2 || ^0.7 || ^1.0",
-    "wp-coding-standards/wpcs": "dev-develop#fca9d9e as 2.3.0",
-    "automattic/vipwpcs": "2.3.3",
+    "wp-coding-standards/wpcs": "dev-develop#bb792cb as 2.3.0",
```
Comment (swissspidy) [inline composer.json]:
```suggestion
    "wp-coding-standards/wpcs": "3.0.0",
```

The commit hash thing was just a workaround we needed before 3.0 was released.

File: composer.json [hunk 2]
```diff
@@ -7,8 +7,8 @@
...
+    "automattic/vipwpcs": "^2.3.3 || ^2.3.4",
```
Comment (swissspidy) [inline composer.json]: Why 2 versions?

Given the desired update to WPCS 3.0 I'd probably just switch to dev-develop here as a new release hasn't been tagged yet.

```suggestion
    "automattic/vipwpcs": "dev-develop",
```

File: composer.json (hunk 1 above)
Comment (mukeshpanchal27) [reply composer.json]: Make sense. Update WPCS to 3.0 and `automattic/vipwpcs` to last commit hash for development branch.

**Review by swissspidy — APPROVED** [review]

File: composer.json [hunk 3]
```diff
@@ -39,20 +38,12 @@
       "dealerdirect/phpcodesniffer-composer-installer": true,
       "composer/installers": true,
       "phpstan/extension-installer": true,
-      "cweagans/composer-patches": true
+      "cweagans/composer-patches": false
```
Comment (felixarntz) [inline composer.json:41]: Shouldn't we remove this entry instead of setting it to `false`, given we're no longer using the package?

**Review by felixarntz — APPROVED** [review]
@mukeshpanchal27 Overall LGTM, one minor question.

File: composer.json (hunk 3 above)
Comment (swissspidy) [reply composer.json:41]: The plugin could still be used by one of the other dependencies. Explicitly setting this to false prevents it from running unexpectedly. So this is good from a security point of view.

**Pull request merged**